src/runge_kutta.jl

```diff
@@ -0,0 +1,396 @@
+# Butcher Tableaus
+# see Hairer & Wanner 1992, p. 134, 166
+
+abstract Tableau{Name, S, T<:FloatingPoint}
+# Name is the name of the tableau/method (a symbol)
+# S is the number of stages (an int)
+# assumes fields
+# order::(Int...) # order of the method(s)
+# a::Matrix{T}  # SxS-1 matrix
+# b::Matrix{T}  # 1 or 2 x S matrix (fixed step/ adaptive)
+# c::Vector{T}  # S
+#
+# For a tableau:
+#  c1  | a_11   ....   a_1s
+#  .   | a_21 .          .           
+#  .   | a_31     .      .           
+#  .   | ....         .  .           
+#  c_s | a_s1  ....... a_ss
+# -----+--------------------
+#      | b_1     ...   b_s   this is the one used for stepping
+#      | b'_1    ...   b'_s  this is the one used for error-checking
+
+Base.eltype{N,S,T}(b::Tableau{N,S,T}) = T
+
+# Test whether it's an explicit method
+isexplicit(b::Tableau) = istril(b.a)
+isadaptive(b::Tableau) = size(b.b, 1)==2
+order(b::Tableau) = b.order
+
+# The advantage of each having its own type, makes it possible to have
+# specialized methods for a particular tablau
+immutable TableauRKExplicit{Name, S, T} <: Tableau{Name, S, T}
+    order::(Int...) # the order of the methods
```

Comment:
This will break on 0.4 with the newly introduced tuple changes; at the moment, it should probably be `order::(@compat(Tuple{Vararg{Int}}))` but I'm unsure on how this works in this context; the outer parenthesis might throw this off.

It will work on 0.3 as is, though :)